Fix Raspberry Pi 2, Pi3 support

Yackou opened the pull request:
- Fix Raspberry Pi model detection
- Add model detection in GPIO module
- Try to fix mailbox char device detection on various kernels
- Fix compiler warnings

Tested on a Raspberry Pi 2, with both GPIO and PWM modules, on a 3.18.7 kernel.

Yackou commented:
inspired from:
* [tylerwowen/pi2](https://github.com/tylerwowen/RPIO/tree/tylerwowen-pi2)
* [sarfata/pi-blaster#49](https://github.com/zador-blood-stained/pi-blaster/commit/0d063223bf3bcd8687ebb7deab0f93e1e24883cc)

dalanmiller commented:
Since this hasn't been merged yet, @Yackou would you mind adding Raspberry Pi 3?

Yackou commented:
I'll try to have a look, but not before at least 2 weeks (currently away
from home), and I won't be able to test it myself since I only have a Pi2.

Le 3 août 2016 12:25 AM, "Daniel Alan Miller" <notifications@github.com> a
écrit :

> Since this hasn't been merged yet, @Yackou <https://github.com/Yackou>
> would you mind adding Raspberry Pi 3?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/metachris/RPIO/pull/82#issuecomment-237065584>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AG9pNyxgG_hUyvv_RztR3dXJ8fwlJypRks5qb8O8gaJpZM4I9T3E>
> .
>

Yackou committed:
Fix Raspberry Pi2, Pi3 support

. Fix Raspberry Pi model detection
. Add model detection in GPIO module
. Try to fix mailbox char device detection on various kernels
. Fix compiler warnings

Yackou commented:
I have updated the pull request with support for Raspberry Pi3 (based on [limuxy/master](https://github.com/limuxy/RPIO/commit/b2c997a3be4bef27cb110fda3cbe7d52239c0ded)) but I don't have one to test.
@dalanmiller could you test whether this works for you?

K1773R commented:
Could you test the PWM too?

ac3d912 commented:
So, I tested it on my RPi 3... pretty sure this is wrong... should be 'a22082'
+    'a02082': ('3B', '1.0', 1024, '?')

cat /proc/cpuinfo
...
Hardware        : BCM2709
Revision        : a22082

doom-fr commented:
Thanks for this fork.
I have tried it. I have made some changes to make it works
(first of all adding my board ( Hardware	: BCM2709 and Revision	: a02082))
It works but sometimes, the raspberry is completely frozen.
Do you think it can came from the RPIO module ?

Thx.

DEvil0000 commented:
It could if you use channels used by drivers or the OS. There was a issue somewhere here listing some channels you should not use. But be carefull - the list could be outdated.

Yackou committed:
get_cpuinfo_revision fixes

. Distinguish board revision from CPU model.
. Consider the board revision to be 3 (latest) by default.
. Use the CPU model (and not the board revision) to determine the GPIO base
address.
. Use the board revision to determine which pin/gpio mappings to use.
. Fix the pin to gpio mapping definition for revision 3 boards.
. Add all known models (according to http://elinux.org/RPi_HardwareHistory)
to the MODEL_DATA (sysinfo) definition.
. Only use the Pi revision (and no longer the sysinfo string) to determine
which GPIO list to use in the Python scripts.

Yackou commented:
FWIW, I have tried making the board detection a bit more generic and robust based on the comments in this thread.
The last update still works fine on my Pi2, feel free to test on your boards as well.

@doom-fr : freezes often come from power supply issues (at least based on my experience, as I greatly reduced freezes by using a more powerful supply), and I believe this is even more true on Pi2 and Pi3 boards because of the increased power requirements. Since the purpose of RPIO is to drive GPIOs, it may also make the freezes more or less frequent depending on the load you put on the GPIOs.

TafThorne commented:
I'll try and get this code onto my Raspberry Pi 3 Model B V1.2 and test it.  I shall report back in a few hours.

metachris commented:
Looks good. @TafThorne thanks for checking, I'm interested to hear what you find out! 

I'm sorry that I have so little time for RPIO at the moment. If you guys are interested in maintaining RPIO I'll give you write access.

TafThorne commented:
This is the first bit of Python work I have ever done so I am probably not a great candidate for maintaining the repository.  I am probably going to spend the next hour working out how to get from source code to the file that ends up at http://pypi.python.org/packages/source/R/RPIO/RPIO-0.10.0.tar.gz so that I can inject that into my Yocto build and run it on my Pi 3 (without going via the server).

TafThorne commented:
Cool.  I have followed https://packaging.python.org/tutorials/distributing-packages/ and produced a RPIO-2.0.0b1.tar.gz which looks to contain the same files as are present in the RPIO-0.10.0.tar.gz I mnetioned.  Now I just have to bundle that file into my Yocto script instead of the reportedly sourced one and I can test it out on the Pi 3 I have.  If anyone has some suggestions of how I could possibly make this more convoluted please let me know.

metachris commented:
FYI - this is how you can build and test the GPIO and PWM parts on the Pi: https://github.com/metachris/RPIO/blob/master/fabfile.py#L104

To run this automatically on your Pi (which you can access on the network), update `fabfile.py` with the correct host, then you can execute this command:

    $ fab clean upload build test_gpio test3_gpio test_pwm

DEvil0000 commented:
I am basically interested but I do not have much time till at least mid september.

Von meinem Samsung Gerät gesendet.

-------- Ursprüngliche Nachricht --------
Von: Chris Hager <notifications@github.com> 
Datum: 21.07.2017  10:32  (GMT+01:00) 
An: metachris/RPIO <RPIO@noreply.github.com> 
Cc: "A. Binzxxxxxx" <alexander@binzberger.de>, Comment <comment@noreply.github.com> 
Betreff: Re: [metachris/RPIO] Fix Raspberry Pi 2, Pi3 support (#82) 

Looks good. @TafThorne thanks for checking, I'm interested to hear what you find out!
I'm sorry that I have so little time for RPIO at the moment. If you guys are interested in maintaining RPIO I'll give you write access.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/metachris/RPIO","title":"metachris/RPIO","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/metachris/RPIO"}},"updates":{"snippets":[{"icon":"PERSON","message":"@metachris in #82: Looks good. @TafThorne thanks for checking, I'm interested to hear what you find out! \r\n\r\nI'm sorry that I have so little time for RPIO at the moment. If you guys are interested in maintaining RPIO I'll give you write access."}],"action":{"name":"View Pull Request","url":"https://github.com/metachris/RPIO/pull/82#issuecomment-316940154"}}}​

JamesGKent commented:
i've taken @Yackou 's fork and added support for all of the boards listed [here](http://elinux.org/RPi_HardwareHistory), my fork is [here](https://github.com/JamesGKent/RPIO/tree/v2), but don't want to issue a pull request until I've tested on all the boards i have (pi 1b v1.1, pi2, pi3 and pi zero w) particularly the pwm.

metachris commented:
Great, thanks for testing, in particular the pwm. 

You could of course already open a pull request and mark it as "work in progress" and add commits as necessary.

TafThorne commented:
Sorry it took me a while to get back.  I am happy to report that I have had RPIO v2.0.0-beta-1 working on a Raspberry Pi 3 Model B V1.2.  I have tested that I can drive all of the pins high or low on demand.  I have not tested the ability to read input on the pins yet.  I do not have any other versions of Raspberry Pi to test with.

DEvil0000 commented:
Tested basic functionality. Works for me on PI 3 revision a02082 from @JamesGKent
